src/snowflake/connector/cursor.py

```diff
@@ -929,7 +1141,9 @@ def execute_async(self, *args: Any, **kwargs: Any) -> dict[str, Any]:
         kwargs["_exec_async"] = True
         return self.execute(*args, **kwargs)
 
-    def describe(self, *args: Any, **kwargs: Any) -> list[ResultMetadata]:
+    def describe(
+        self, *args: Any, **kwargs: Any
+    ) -> list[ResultMetadata | ResultMetadataV2]:
```

Comment:
Shouldn't this be just List[ResultMetadata]?

Reply:
Thanks for catching this: it should be just `List[ResultMetadata]` since describe only returns the old format. Same for `_describe_internal` in your other comment.